addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2031,6 +2074,7 @@ const SnippetOptionWidget = Widget.extend({
                     // property we are editing, nothing more has to be done.
                     // (except adding the extra class)
                     this.$target.addClass(params.extraClass);
+                    if (previewMode === false) await this.updateChangesInWysiwyg();
```

Comment:
I agree with you and that's what I had done. @Goaman changed that to fix a bug, maybe he can explain it here.

Reply:
There is a `return` just after the instruction that was `this.$target.addClass(params.extraClass);`. Without sending the change to the wysiwyg here, it would not be done.

Reply:
Yes I understand that, the question was: why can't it be done at the end of the `_select` method, which calls all those methods where you added the calls to `updateChangesInWysiwyg`. But I guess we'll see with the `observeChanges` stuff

Reply:
The reason is that this strategy does not work in all cases.

Reply:
I guess but the question was "why?" ;)

Reply:
@Goaman I'm with @qsm-odoo and @Zinston on this. I do not understand either. We can schedule a call together if it's easier for you to explain.

Reply:
@Goaman @dmo-odoo @qsm-odoo  Did you guys get around to discussing this in more detail? Did you make sense of it?

Reply:
I don't know exactly which options were broken.
This method erase all the references of the child contained in the option `$target` which might confuse options.
As the method is destructive, I used it only when references in the DOM were changed.

As the mutation method will not be destructive, it will be much safer to wrap it around the `_select` method.